applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+
+    index = model_part.NumberOfConditions()
+    for condition in base_model_part.Conditions:
+        if condition.Is(Kratos.PERIODIC):
+            index += 1
+            node_id_list = [node.Id for node in condition.GetNodes()]
+            periodic_condition = model_part.CreateNewCondition(
+                "PeriodicCondition", index, node_id_list, properties)
+            periodic_condition.Set(Kratos.PERIODIC)
+
+
+def CreateSteadyScalarScheme(relaxation_factor):
+    return steady_scalar_scheme(relaxation_factor)
+
+
+def CreateSteadyAlgebraicFluxCorrectedScheme(relaxation_factor,
+                                                     boundary_flags,
+                                                     is_periodic):
+    if (is_periodic):
+        return afc_steady_scalar_scheme(relaxation_factor, boundary_flags,
+                                        KratosCFD.PATCH_INDEX)
+    else:
+        return afc_steady_scalar_scheme(relaxation_factor, boundary_flags)
+
+
+def CreateBossakRelaxationScalarScheme(bossak_value, relaxation_factor, scalar_variable,
+                             scalar_rate_variable,
+                             relaxed_scalar_rate_variable):
+    return bossak_scheme(bossak_value, relaxation_factor, scalar_variable,
+                         scalar_rate_variable, relaxed_scalar_rate_variable)
+
+
+def GetBoundaryFlags(boundary_flags_parameters):
+    if (boundary_flags_parameters.size == 0):
+        raise Exception("No boundary flags were found")
+
+    flags = Kratos.KratosGlobals.GetFlag(
+        boundary_flags_parameters[0].GetString())
+    for i in range(1, boundary_flags_parameters.size()):
+        flags |= Kratos.KratosGlobals.GetFlag(
+            boundary_flags_parameters[i].GetString())
+
+    return (flags)
+
+
+def GetDefaultConditionName(model_part):
+    process_info = model_part.ProcessInfo
+    if (process_info.Has(Kratos.DOMAIN_SIZE)):
+        domain_size = process_info[Kratos.DOMAIN_SIZE]
+        if (domain_size == 2):
+            return "LineCondition"
+        elif (domain_size == 3):
+            return "SurfaceCondition"
+        else:
+            raise Exception("Unsupported domain size. [ DOMAIN_SIZE = " +
+                            str(domain_size) + " ].")
+    else:
+        raise Exception("DOMAIN_SIZE is not found in process info of " +
+                        model_part.Name() + ".")
+
+
+def GetConvergenceInfo(variable,
+                       relative_error,
+                       relative_tolerance,
+                       absolute_error=-1.0,
+                       absolute_tolerance=-1.0):
+    info = "[ Obtained ratio: {0:6e}; Expected ratio: {1:6e}".format(
+        relative_error, relative_tolerance)
+    if (absolute_error >= 0.0 and absolute_tolerance >= 0.0):
+        info += "; Absolute norm: {0:6e}; Expected norm: {1:6e}".format(
+            absolute_error, absolute_tolerance)
+    info += " ] - " + str(variable.Name())
+    return info
```

Comment:
This is to output the steadiness convergence right?

Reply:
yes. Its merely used to show convergence.